src/mapping/PetRadialBasisFctMapping.hpp

```diff
@@ -1436,11 +1414,8 @@ PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::bgPreallocationMatrixA(mesh::
     }
 
     // -- PREALLOCATE THE COEFFICIENTS --
-    results.clear();
-    auto search_box = query::getEnclosingBox(oVertex, supportRadius);
-
-    tree->query(bg::index::intersects(search_box) and bg::index::satisfies([&](size_t const i) { return bg::distance(oVertex, inMesh->vertices()[i]) <= supportRadius; }),
-                std::back_inserter(results));
+    query::Index indexTree(inMesh);
+    auto         results = indexTree.getVerticesInsideBox(oVertex, supportRadius);
```

Comment:
It would be interesting to benchmark the inner-loop vs outer-loop version. Not too important here though.